opentelemetry-instrumentation/src/opentelemetry/instrumentation/_blobupload/api/blob.py

```diff
@@ -0,0 +1,125 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import json
+from types import MappingProxyType as _frozendict
+from typing import Mapping, Optional
+
+
+class Blob:
```

Comment:
If you implemented this as a dataclass (perhaps frozen), you could avoid having to implement:
* most of `__init__`
* the three `@property` methods
* `__eq__` probably
* `__repr__`